Library/Formula/libmill.rb

```diff
@@ -0,0 +1,29 @@
+class Libmill < Formula
+  desc "Go-style concurrency in C"
+  homepage "http://libmill.org/"
+  head "https://github.com/sustrik/libmill.git"
+  url "http://libmill.org/libmill-0.16-beta.tar.gz"
+  sha256 'c6d68d8291dadf2ee111f6ceed057f2fd267e7dc334bef8d15b6316c35b6a0ca'
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "cmake" => :build
+  env :std
+
+  def install
+    if build.head?
+      system "./autogen.sh"
+    end
+    if ENV.compiler == :clang
+      ENV.append_to_cflags "-O1"
+    end
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+    system "make", "check"
+    system "make", "install"
+  end
+
+  test do
```

Comment:
make check runs libmill's unit tests. Those unit tests must be run from within the project directory.

I did not see a way to run 'make check' from test do.